pages/home.jsx

```diff
@@ -146,11 +149,121 @@ var BlogSection = React.createClass({
   }
 });
 
+var validateSignupForm = function(signUpFormState) {
+  var errors = [];
+  // regex copied from http://stackoverflow.com/a/46181
+  var regexEmail = /^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$/i;
+  if (!regexEmail.test(signUpFormState.email)) {
+    errors.push("Please enter an email address.");
+  }
+
+  return errors;
+};
+
+var ModalPledge = React.createClass({
+  mixins: [React.addons.LinkedStateMixin],
+  getInitialState: function() {
+    return {
+      email: "",
+      validationErrors: []
+    };
+  },
+  handleSubmit: function(e) {
+    var validationErrors = validateSignupForm(_.pick(this.state,"email"));
+
+    if (validationErrors.length) {
+      e.preventDefault();
+      this.setState({validationErrors: validationErrors});
+      return;
+    }
+
+    if (process.env.NODE_ENV !== 'production' &&
+        !process.env.PLEDGE_MAILINGLIST_URL) {
+      e.preventDefault();
+      alert("PLEDGE_MAILINGLIST_URL is not defined. Simulating " +
+            "a successful pledge signup now.");
+      window.location = "?pledge=thanks";
+    }
+  },
+  renderValidationErrors: function() {
+    if (this.state.validationErrors.length) {
+      return (
+        <div className="alert alert-danger">
+          <p className="error-msg">Please enter an email address.</p>
```

Comment:
Cool, I'm tempted to factor out a `<EmailSignupForm>` component actually... but since my work for #900 is in progress already and it involves copying a signup form to hero unit section, I would wanna wait until both 2 PRs are landed first to save me from fixing merge conflicts.  Ticket created: https://github.com/mozilla/teach.webmaker.org/issues/1050


Unit tests ticket created: https://github.com/mozilla/teach.webmaker.org/issues/1049